src/core/tests/hostname_test.cpp

```diff
@@ -10,58 +10,68 @@
 namespace dsn {
 namespace replication {
 
+TEST(ip_to_hostname, ipv4_validate)
+{
+    rpc_address rpc_test_ipv4;
+    struct ip_test
+    {
+        std::string ip;
+        bool result;
+    } tests[11] = {{"127.0.0.1:8080", 1},
+                   {"172.16.254.1:1234", 1},
+                   {"172.16.254.1:222222", 0},
+                   {"172.16.254.1", 0},
+                   {"2222,123,33,1:8080", 0},
+                   {"123.456.789.1:8080", 0},
+                   {"001.223.110.002:8080", 0},
+                   {"172.16.254.1.8080", 0},
+                   {"172.16.254.1:8080.", 0},
+                   {"127.0.0.11:123!", 0},
+                   {"127.0.0.11:123!", 1}};
+
+    for (int i = 0; i < 11; i++) {
```

Comment:
for (auto test : tests) {
    ASSERT_EQ(rpc_test_ipv4.from_string_ipv4(test.ip.c_str()), test.result);
}